patient-appointment-management/functions/helpers.private.js

```diff
@@ -271,7 +301,175 @@ async function getParam(context, key) {
   }
 }
 
+/*
+ * --------------------------------------------------------------------------------
+ * validates appoointment data to guard against json injection
+ *
+ * param:
+ * - appointment: json of appointment data
+ *     event_type: 'BOOKED',
+ *     event_datetime_utc: null,
+ *     patient_id: '1000',
+ *     patient_first_name: 'Jane',
+ *     patient_last_name: 'Doe',
+ *     patient_phone: test_phone_number,
+ *     provider_id: 'afauci',
+ *     provider_first_name: 'Anthony',
+ *     provider_last_name: 'Fauci',
+ *     provider_callback_phone: '(800) 111-2222',
+ *     appointment_location: 'Owl Health Clinic',
+ *     appointment_id: '20000',
+ *     appointment_timezone: '-0700',
+ *     appointment_datetime: appt_datetime.toISOString(),
+ * --------------------------------------------------------------------------------
+ */
+function validateAppointment(context, appointment) {
```

Comment:
I'm not sure of the value of this function, given that we never (and should never) run `eval()` on any JSON values. If there's a demonstrable JSON injection attack that can be prevented by code along these lines, I would prefer if sanitization were handled by a dedicated library from NPM, rather than code developed in-house.

Reply:
this function was added per Manuel from security's recommendation to implement some validation of format of json elements. Not sure if it requires changing in this version, can you point us to good sanitization packages? thx

Reply:
Sanitization and validation are very different processes. Sanitization would prevent JSON injection attacks (which the comment for this function seems to indicate is the goal here) -- most attacks around that are prevented by us not using `eval()`, but a package that can handle extra sanitization if needed is [ajv-sanitizer](https://www.npmjs.com/package/ajv-sanitizer).

Validation involves checking that the values inside the object that you've parsed from the JSON you consumed are of an expected type, and that each value upholds certain properties (IE, does a string look like a valid external email). For that, please see Kelley's comment regarding Twilio's Lookup API -- it's our standard for bulletproof validation. In case that API can't work in this role for any reason, I can recommend:

* `ajv-sanitizer`, mentioned above, which supports validating via a schema
* [is-my-json-valid](https://github.com/mafintosh/is-my-json-valid), a similar package
* [validator.js](https://github.com/validatorjs/validator.js), which is a set of small composable utility functions (like `isEmail()`) and is much simpler than the first two options

Since, in practice, this will likely be a relatively quick change to something with a big security impact (custom validation/sanitization code is a bit of a security red flag), I would prefer to see the change made as part of this PR.